ci-operator/step-registry/fips-check/optional-operators/fips-check-optional-operators-commands.sh

```diff
@@ -0,0 +1,150 @@
+#!/bin/bash
+set -o nounset
+set -o errexit
+set -o pipefail
+
+
+function set_proxy () {
+    if test -s "${SHARED_DIR}/proxy-conf.sh" ; then
+        echo "setting the proxy"
+        # cat "${SHARED_DIR}/proxy-conf.sh"
+        echo "source ${SHARED_DIR}/proxy-conf.sh"
+        source "${SHARED_DIR}/proxy-conf.sh"
+    else
+        echo "no proxy setting."
+    fi
+}
+
+function run_command() {
+    local CMD="$1"
+    echo "Running Command: ${CMD}"
+    eval "${CMD}"
+}
+
+set_proxy
+pass=true
+# get a master node
+master_node_0=$(oc get node -l node-role.kubernetes.io/master= --no-headers | grep -Ev "NotReady|SchedulingDisabled"| awk '{print $1}' | awk 'NR==1{print}')
+if [[ -z $master_node_0 ]]; then
+    echo "Error master node0 name is null!"
+    pass=false
+fi
+# create a ns
+project="optional-scan-$RANDOM"
+run_command "oc new-project $project --skip-config-write"
+if [ $? == 0 ]; then
+    echo "create $project project successfully" 
+else
+    echo "fail to create $project project."
+    pass=false
+fi
+# skip test for disconnected clusters
+oc label namespace "$project" security.openshift.io/scc.podSecurityLabelSync=false pod-security.kubernetes.io/enforce=privileged pod-security.kubernetes.io/audit=privileged pod-security.kubernetes.io/warn=privileged --overwrite=true || true
+cluster_http_proxy=$(oc get proxy cluster -o=jsonpath='{.spec.httpProxy}')
+attempt=0
+while true; do
+    out=$(oc --request-timeout=60s -n "$project" debug node/"$master_node_0" -- chroot /host bash -c "export http_proxy=$cluster_http_proxy; curl -sSI ifconfig.me --connect-timeout 5" 2> /dev/null || true)
+    if [[ $out == *"Via: 1.1"* ]]; then
+        echo "This is not a disconnected cluster"
+        break
+    fi
+    attempt=$(( attempt + 1 ))
+    if [[ $attempt -gt 3 ]]; then
+        echo "This is a disconnected cluster, skip testing"
+        oc delete ns "$project"
+        exit 0
+    fi
+    sleep 5
+done
+
+# podman login and export auth for opm
+run_command "oc extract secret/pull-secret -n openshift-config --confirm --to /tmp"; ret=$?
+if [[ $ret -ne 0 ]]; then
+    echo "failed to get the auth"
+    oc delete ns "$project"
+    pass=false
+fi
+auth_dir="/tmp/containers"
+if [ ! -d "${auth_dir}" ]; then
+  mkdir "${auth_dir}" || true
+fi
+mv /tmp/.dockerconfigjson ${auth_dir}/auth.json || true
+echo "check the auth file exists or not: $(ls -ltr ${auth_dir}/auth.json)"
+export XDG_RUNTIME_DIR="${XDG_RUNTIME_DIR:-/tmp}"
```

Comment:
If this `XDG_RUNTIME_DIR` was set before, it won't be set `/tmp/containers/auth.json` value. I guess this leads https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/45225/rehearse-45225-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-debug/1722430610884005888/artifacts/installer-rehearse-debug/fips-check-optional-operators/build-log.txt